apps/core/src/hooks/useGetKioskContents.ts

```diff
@@ -1,71 +1,102 @@
-	obj.data?.content &&
-	'fields' in obj.data.content &&
-	(obj.data.content.fields.for ?? obj.data.content.fields.kiosk);
+	// fetch the user's kiosks
+	const ownedKiosks = await rpc.multiGetObjects({
+		ids: ids,
+		options: {
+			showContent: true,
+		},
+	});
 
-export function useGetKioskContents(address?: SuiAddress | null, disableOriginByteKiosk?: boolean) {
-	const rpc = useRpcClient();
-	const { data } = useGetOwnedObjects(address, {
-		MatchAny: [
-			{ StructType: KIOSK_OWNER_CAP },
-			...(!disableOriginByteKiosk ? [{ StructType: ORIGINBYTE_KIOSK_OWNER_TOKEN }] : []),
-		],
+	// find object IDs within a kiosk
+	const kioskObjectIds = await Promise.all(
+		ownedKiosks.map(async (kiosk) => {
+			if (!kiosk.data?.objectId) return [];
+			const objects = await rpc.getDynamicFields({
+				parentId: kiosk.data.objectId,
+			});
+			return objects.data.map((obj) => obj.objectId);
+		}),
+	);
+
+	// fetch the contents of the objects within a kiosk
+	const kioskContent = await rpc.multiGetObjects({
+		ids: kioskObjectIds.flat(),
+		options: {
+			showDisplay: true,
+			showType: true,
+		},
 	});
 
-	// find list of kiosk IDs owned by address
-	const kioskIds = data?.pages.flatMap((page) => page.data).map((obj) => getKioskId(obj)) ?? [];
+	return kioskContent;
+}
 
-	return useQuery({
-		queryKey: ['get-kiosk-contents', address, kioskIds],
-		queryFn: async () => {
-			if (!kioskIds.length) return [];
+async function getSuiKioskContents(address: SuiAddress, rpc: JsonRpcProvider) {
+	const ownedKiosks = await getOwnedKiosks(rpc, address!);
+	const kioskContents = await Promise.all(
+		ownedKiosks.kioskIds.map(async (id) => {
+			return fetchKiosk(rpc, id, { limit: 1000 }, {});
+		}),
+	);
+	const items = kioskContents.flatMap((k) => k.data.items);
+	const ids = items.map((item) => item.objectId);
 
-			// fetch the user's kiosks
-			const ownedKiosks = await rpc.multiGetObjects({
-				ids: kioskIds,
-				options: {
-					showContent: true,
-				},
-			});
+	// fetch the contents of the objects within a kiosk
+	const kioskContent = await rpc.multiGetObjects({
+		ids,
+		options: {
+			showContent: true,
+			showDisplay: true,
+			showType: true,
+		},
+	});
 
-			// find object IDs within a kiosk
-			const kioskObjectIds = await Promise.all(
-				ownedKiosks.map(async (kiosk) => {
-					if (!kiosk.data?.objectId) return [];
-					const objects = await rpc.getDynamicFields({
-						parentId: kiosk.data.objectId,
-					});
-					return objects.data.map((obj) => obj.objectId);
-				}),
-			);
+	return kioskContent;
+}
 
-			// fetch the contents of the objects within a kiosk
-			const kioskContent = await rpc.multiGetObjects({
-				ids: kioskObjectIds.flat(),
-				options: {
-					showDisplay: true,
-					showType: true,
-				},
-			});
+export function useGetKioskContents(address?: SuiAddress | null, disableOriginByteKiosk?: boolean) {
+	const rpc = useRpcClient();
+	return useQuery({
+		// eslint-disable-next-line @tanstack/query/exhaustive-deps
```

Comment:
wasn't sure about this one - was complaining about `rpc` being required in the query key, but that can't be serialized in our cache

Reply:
Yeah, I was chatting with @Jordan-Mysten about this a few weeks ago.

React-query doesn't allow non-serializable query keys, and the linter isn't smart enough to take `rpc.connectionName` or whatever the field for the connection URL is. Besides that though, we don't really treat the active network connection as UI state, and as such we don't want to cache queries based on the active network connection. So, I think this is the correct approach